Sources/Action.swift

```diff
@@ -18,12 +18,19 @@ import Result
 ///
 /// `Action` enforces serial execution, and disables the `Action` during the execution.
 public final class Action<Input, Output, Error: Swift.Error> {
-	private let deinitToken: Lifetime.Token
+	private enum Availability {
+		case enabledIdle
+		case enabledExecuting
+		case disabledIdle
+		case disabledExecuting
```

Comment:
This is really modeling two independent boolean states. Can we separate these and add them to `ActionState` individually? I think that will simplify the code a bit as well.